src/USER-CAC/cg_blas.h

```diff
@@ -0,0 +1,82 @@
+/* If the BLAS are not installed, then the following definitions
```

Comment:
we bundle a basic BLAS/LAPACK package in `lib/linalg`, which is a subset of BLAS/LAPACK with only the required functions. So code can assume, that BLAS/LAPACK is always available. CMake will search for optimized BLAS first and then fall back to compiling the bundled library. for conventional build, the user has to read the docs,

Reply:
Honestly not sure whether to just completely delete this possibility to use BLAS (since my problem dimensions are n=2, or n=3 and are likely not going to benefit anyway) or to leave it suggested in the docs in case someone happens to implement something that complicated and needs it. Is there a way to automate a check for whether BLAS is installed or not for the traditional build method?

Reply:
There is no automated check in the conventional build. We have things set up that the default build uses the library from lib/linalg. You could put a folder `cac` into `lib` which works similar to others with a selection of pre-configured templates for `Makefile.lammps`. Or switch from BLAS to using `Eigen` like the USER-SMD package does (see `lib/smd`). Or just have replacement code inserted directly. Simple benchmarking should tell you whether the optimization of BLAS is really needed. For matrices with small ranks (< 10) a fully unrolled inline code (no pipeline stalling loops or ifs) can be much faster.

Reply:
Yea I honestly can't foresee an application where the dimension will be higher than three for the uses in this package anyway. I'll probably just get rid of the BLAS related stuff, leaving the existing unrolled loops that have been used by default, and worst case scenario someone can just nab the code from the vanilla asa_cg algorithm again when they need that.